app/views/partials/paymentAllocations/PaymentAllocationsForNonLpi.scala.html

```diff
@@ -39,16 +38,15 @@
         <tr>
             <td class="govuk-table__cell" id="payment-allocation-@index">
                 <a class="govuk-link" style="display:block" href="@paymentAllocationHref(paymentAllocation, isAgent)">
-                    @messages(paymentAllocation.allocationDetail.getOrElse(throw MissingFieldException("Payment Allocation Detail")).getPaymentAllocationKeyInPaymentAllocations)
-                    <span class="govuk-visually-hidden">@messages("paymentAllocation.taxYear", (paymentAllocation.allocationDetail.getOrElse(throw MissingFieldException("Payment Allocation Detail")).getTaxYear(implicitDateFormatter) - 1).toString, paymentAllocation.allocationDetail.get.getTaxYear(implicitDateFormatter).toString)</span>
+                    @messages(paymentAllocation.validatedAllocationDetail.getPaymentAllocationKeyInPaymentAllocations)
```

Comment:
same here, if you decided to go via path of introducing validation method, 
should be ok to add it on the level of controller as we do not have a view model at the moment